lib/message_request.js

```diff
@@ -0,0 +1,175 @@
+var logger = require('minilog')('message:request');
+
+var opTable = {
+  control: ['nameSync'],
+  message: ['publish', 'subscribe', 'sync', 'unsubscribe'],
+  presence: ['get', 'set', 'subscribe', 'sync', 'unsubscribe'],
+  status: ['get', 'set', 'subscribe', 'sync', 'unsubscribe'],
+  stream: ['get', 'push', 'subscribe', 'sync', 'unsubscribe']
+};
+
+var Request = function (operation, scope) {
+  this.message = {
+    op: operation,
+    to: scope
+  };
+
+  if (!this._isValid()) {
+    logger.error('invalid request. op: ' + this.op + '; to: ' + this.to);
+  }
+};
+
+Request.buildGet = function (scope, options) {
+  var request = new Request('get', scope);
+  if (request) {
+    request.setOptions(options);
+  }
+
+  return request;
+};
+
+Request.buildPublish = function (scope, value) {
+  var request = new Request('publish', scope);
+  if (request) {
+    request.setAttr('value', value);
+  }
+
+  return request;
+};
+
+Request.buildPush = function (scope, resource, action, value) {
+  var request = new Request('push', scope);
+  if (request) {
+    request.setAttr('resource', resource);
+    request.setAttr('action', action);
+    request.setAttr('value', value);
+  }
+
+  return request;
+};
+
+Request.buildNameSync = function (scope, options) {
+  var request = new Request('nameSync', scope);
+  if (request) {
+    request.setOptions(options);
+  }
+
+  return request;
+};
+
+Request.buildSet = function (scope, value, key, userType) {
+  var request = new Request('set', scope);
+
+  if (request) {
+    request.setAttr('value', value);
+    request.setAttr('key', key);
+    request.setAttr('type', userType);
+  }
+
+  return request;
+};
+
+Request.buildSync = function (scope) {
+  return new Request('sync', scope);
+};
+
+Request.buildSubscribe = function (scope) {
+  return new Request('subscribe', scope);
+};
+
+Request.buildUnsubscribe = function (scope, options) {
+  return new Request('unsubscribe', scope);
+};
+
+// TO DO: config class should return the required data via getters
+Request.prototype.setAuthData = function (configuration) {
+  if (configuration && configuration.auth) {
+    this.setAttr('auth', configuration.auth);
+    this.setAttr('userId', configuration.userId);
+    this.setAttr('userType', configuration.userType);
+    this.setAttr('accountName', configuration.accountName);
+  }
+};
+
+Request.getAttr = function (message, attr) {
+  return (message && message[attr]);
+};
+
+// Instance methods
+
+Request.prototype.getMessage = function () {
+  return this.message;
+};
+
+Request.prototype.setOptions = function (options) {
+  if (options) {
+    this.message.options = options;
+    this.message.options.version = 2;
+  } else {
+    this.message.options = { version: 1 };
```

Comment:
No. we should not have version: 1 anymore at all. And all Presence Sync messages should have `{ version: 2 }` attached. This is the wrong way to do this and it's only here because we need to support this in this existing way.